packages/eslint-plugin-gestalt/src/no-box-disallowed-props.js

```diff
@@ -4,6 +4,49 @@
 
 // @flow strict
 const allowedBaseProps = [
+  // React / DOM
+  'id',
```

Comment:
This list contains props we found in the Pinterest codebase & ones referenced in https://github.com/jsx-eslint/jsx-ast-utils/blob/c554f28b6b1deaf57d6e244aeac3bd7fc6c31039/src/eventHandlers.js#L16 (do know that some of these don't make sense on `div` / `main` / ..., so I didn't include all of them)